### PR TITLE
Be more data driven

### PR DESCRIPTION
**Pull request by janhohenheim**
This code should make it more clear that the asset key can be loaded from disk. 
I tried to limit my use of iterators in this PR because they tend to be complicated for beginners, but now there is a bit of a mix between procedural and iterator code. If you want, I can replace it all with procedural code.

Requesting a review from @alice-i-cecile as well.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Review by benfrankel — COMMENTED** [review]
If the keys are always paths, I'm wondering if `asset_server.load(ImagePaths::DUCKY)` could replace `image_handles[&ImageHandles::KEY_DUCKY]`? And if not, why not?

I know SFX require a `Vec`, but technically wouldn't it be even more data-driven if it was a single `Handle<Vec<AudioSource>>` somehow? Then the _number of variations_ would also be data-driven.

**Comment by janhohenheim** [comment]
@benfrankel that would work if you wrap `Vec<AudioSource>>` in a type that implements `Asset`. I can try and see how it feels.

**Comment by janhohenheim** [comment]
Update: if I add a custom resource like `SfxSources(Vec<AudioSource>)`, it would require creation by moving all SFX `AudioSource`s out of `Assets` and into `SfxSources`, which only works *after* the SFX sources are done loading, so we need add an event reader for `AssetEvent` and move there.
That, or we create an own `AssetLoader` with its own logic for reading all files, which is not pretty.

Given the amount of complexity this would add, I'm voting for keeping it as-is. We could do `asset_server.load(FOO_KEY)` for non-SFX, but I don't want to introduce inconsistency.

File: src/assets.rs [hunk 1]
```diff
@@ -25,22 +25,23 @@ pub(super) fn plugin(app: &mut App) {
...
-                "images/ducky.png",
-                |settings: &mut ImageLoaderSettings| {
-                    settings.sampler = ImageSampler::nearest();
-                },
-            ),
-        )]
-        .into();
+
+        let pixel_art_settings = |settings: &mut ImageLoaderSettings| {
+            // Using nearest sampling to preserve pixel art style
```
Comment (benfrankel) [inline src/assets.rs:36]:
```suggestion
            // Use `nearest` image sampling to preserve the pixel art style.
```

File: src/assets.rs [hunk 2]
```diff
@@ -25,22 +25,23 @@ pub(super) fn plugin(app: &mut App) {
...
+            settings.sampler = ImageSampler::nearest();
+        };
+        let mut map = HashMap::new();
+        map.insert(
+            Self::KEY_DUCKY.to_string(),
+            asset_server.load_with_settings(Self::KEY_DUCKY, pixel_art_settings),
+        );
```
Comment (benfrankel) [inline src/assets.rs:43]: This could be the same `[key].into_iter().map().collect()` as the other two structs.

**Review by benfrankel — CHANGES_REQUESTED** [review]

File: src/assets.rs (hunk 1 above)
Comment (benfrankel) [reply src/assets.rs:36]: Or we could mention `ImageFilterMode::Nearest`? That's probably overkill though.

File: src/assets.rs (hunk 2 above)
Comment (janhohenheim) [reply src/assets.rs]: [`f1b7382` (#240)](https://github.com/TheBevyFlock/bevy_quickstart/pull/240/commits/f1b73827958847f8b3759ecccefa02296362b789)

**Review by benfrankel — APPROVED** [review]

**Pull request merged**